src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -1428,7 +1428,14 @@ DatabaseShardImp::setFileStats()
     if (auto const count = numHistoricalShards(lock);
         count >= maxHistoricalShards_)
     {
-        JLOG(j_.warn()) << "maximum number of historical shards reached";
+        if (maxHistoricalShards_)
+        {
+            // In order to avoid excessive output, don't produce
+            // this warning if the server isn't configured to
+            // store historical shards.
+            JLOG(j_.warn()) << "maximum number of historical shards reached";
+        }
+
```

Comment:
If `canAdd_` is true, we can return early.
eg.
```
    if (!canAdd_)
        return;

    auto const count{numHistoricalShards(lock)};
    if (count >= maxHistoricalShards_)
    {
        JLOG(j_.warn()) << "maximum number of historical shards reached";
        canAdd_ = false;
    }
    else if (!sufficientStorage(
                 maxHistoricalShards_ - count,
                 PathDesignation::historical,
                 lock))
    {
        JLOG(j_.warn())
            << "maximum shard store size exceeds available storage space";
    }
```

Reply:
Good idea, done